src/umpire/strategy/DynamicPool.cpp

```diff
@@ -153,9 +158,19 @@ void* DynamicPool::allocateFromResource(std::size_t bytes)
       throw;
     }
   }
+
+  // Add to count
+  m_actual_bytes += bytes;
```

Comment:
I changed the `(de)allocateFromStrategy()` calls to update m_actual_bytes at the source of allocation, to avoid mistakes.